lib/Tumblr/StreamBuilder/Streams/ChronologicalStreamMixer.php

```diff
@@ -118,7 +118,8 @@ protected function mix(
                 throw new \UnexpectedValueException('Unexpected value for sorting option. Options: asc or desc');
             }
         });
-        return new StreamResult(count($res) <= $count, array_slice($res, 0, $count));
```

Comment:
how did this ever work??

Reply:
so yeah, the way we do pagination in some legacy places within tumblr is to overfetch by 1, and that gives us enough of an indication that there's a "next page" (whether the stream is exhausted or not).

you have 100 rows in the database, and you want to paginate by 20, so you fetch 21 per page. the fact that there's a 21st thing on page 1 means there's a page 2, so you have not reached the end of the table, therefore the stream is not exhausted. the first page fetches items 1 through 21, the next page fetches 20 through 31, etc, until the last page tries to fetch 90 through 101, but there is no 101, therefore the stream is actually exhausted.

that's why this works. 😄

Reply:
but...... we need to do a better job of explaining this logic at the framework level, for sure. because otherwise, this PR would break what we're doing in our implementations of these streams.

Reply:
It is also worthwhile to note, in documentation somewhere, that when the `is_exhausted` flag is set to true. It is safe to believe that it is truly exhausted. However, when it is false, the stream may actually be exhausted, but you won't find out until the next pagination. There are some streams, which cannot accurately predict whether they are exhausted. Generally because enumeration has side effects (gatekeeper was like this iirc; not sure if it is still used internally)

Reply:
@bnu0 it's more than great to see you here

Reply:
@lucila @cmantas i agree that it sounds like we should update some docs here, maybe add some inline code comments right here, to explain the thinking behind this code

Reply:
this change sounds good to me. The implementation of ChronologicalStreamMixer shouldn't rely on tricks of inner streams. Right now, the ChronologicalStreamMixer returns is_exhausted=true if we ask only 10 elements and we have only a stream that can_enumerate and that one returns exactly 10 elements (this is what we asked for...).

Reply:
@cyle as an alternative, we can modify line 89 to: 

```
$stream_results = $stream->enumerate($count +1, $cursor->cursor_for_stream($stream), $tracer, $option);
```